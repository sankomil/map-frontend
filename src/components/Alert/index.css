.alert-element {
  position: absolute;
  z-index: 10;
  bottom: 0;
  margin-bottom: 50;
  left: 45vw;
  justify-content: center;
  display: flex;
}

.alert-element > div {
    margin-bottom: 50px;
}