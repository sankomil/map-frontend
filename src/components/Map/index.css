.map-container {
  height: 100vh;
  width: 100vw;
  position: relative;
}
