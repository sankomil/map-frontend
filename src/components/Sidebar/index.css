.sidebar-container {
  position: fixed;
  top: 5%;
  left: 0;
  width: 300px;
  border-radius: 0 50px 50px 0;
  z-index: 10;
  background-color: #fff;
  transition: 0.5s;
}

.sidebar-container-close {
  left: -400px !important;
}

.toggle-container-close {
  left: 0px !important;
}

.toggle-container {
  height: 40px;
  width: 25px;
  position: absolute;
  top: calc(5% + 70px);
  left: 300px;
  border-left: 1px solid #dadce0;
  border-radius: 0 8px 8px 0;
  background: #fff 7px center/7px 10px no-repeat;
  z-index: 10;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  transition: 0.5s;
}

@media screen and (min-width: 950px) {
  .sidebar-container {
    width: 400px;
  }

  .toggle-container {
    left: 400px;
  }

  .sidebar-container-close {
    left: -400px;
  }
}
