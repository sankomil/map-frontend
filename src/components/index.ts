export * from "./Sidebar";
export * from "./MapPin";
export * from "./Map";
