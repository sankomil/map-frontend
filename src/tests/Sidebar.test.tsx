import React from "react";
import { Sidebar } from "../components";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

const getPathToken = jest.fn();

jest.mock("../hooks", () => {
  return {
    useGetRoute: jest.fn(() => ({
      getPathToken,
      loading: false,
      resetAll: jest.fn(),
    })),
    useGlobalContext: jest.fn(() => ({ error: null })),
  };
});

describe("Sidebar test", () => {
  test("Component renders", () => {
    const { container } = render(<Sidebar />);
    expect(container).toMatchSnapshot();
  });

  test("Submit function is called when user submits", () => {
    render(<Sidebar />);
    fireEvent.input(screen.getByTestId("start-point-input"), {
      target: { value: "Test" },
    });
    fireEvent.input(screen.getByTestId("end-point-input"), {
      target: { value: "Test" },
    });
    fireEvent.click(screen.getByTestId("sidebar-submit"));
    expect(getPathToken).toHaveBeenCalled();
  });

  test("User can write in inputs", () => {
    render(<Sidebar />);
    fireEvent.input(screen.getByTestId("start-point-input"), {
      target: { value: "Test" },
    });
    fireEvent.input(screen.getByTestId("end-point-input"), {
      target: { value: "Test" },
    });
    expect(screen.getByTestId("start-point-input")).toHaveValue("Test");
    expect(screen.getByTestId("end-point-input")).toHaveValue("Test");
  });

  test("Users can clear text", () => {
    render(<Sidebar />);
    fireEvent.input(screen.getByTestId("start-point-input"), {
      target: { value: "Test" },
    });
    fireEvent.click(screen.getByTestId("start-point-clear"));
    expect(screen.getByTestId("start-point-input")).toHaveValue("");
  });

  test("Submit button is disabled when no origin and destination is given", () => {
    render(<Sidebar />);

    expect(screen.getByTestId("sidebar-submit")).toHaveAttribute("disabled");
  });
});
