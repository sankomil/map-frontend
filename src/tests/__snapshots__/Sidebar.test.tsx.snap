// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar test Component renders 1`] = `
<div>
  <form>
    <div
      class="d-flex"
    >
      <div
        class="sidebar-container p-4"
        data-testid="sidebar-container"
      >
        <h1
          class="display-5 text-center"
        >
          Route Finder!
        </h1>
        <h4
          class="text-center"
        >
          Now find routes easily.
        </h4>
        <div
          class="d-flex flex-column mt-4"
        >
          <label
            class="mb-1"
            for="start-point"
          >
            Origin:
          </label>
          <div
            class="d-flex align-items-center"
          >
            <input
              class="w-100"
              data-testid="start-point-input"
              id="start-point"
              value=""
            />
            <button
              aria-label="Close"
              class="close btn btn btn-outline-danger ms-3 btn-sm"
              data-testid="start-point-clear"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
        </div>
        <div
          class="d-flex flex-column mt-3"
        >
          <label
            class="mb-1"
            for="end-point"
          >
            Destination:
          </label>
          <div
            class="d-flex align-items-center"
          >
            <input
              class="w-100"
              data-testid="end-point-input"
              id="end-point"
              value=""
            />
            <button
              aria-label="Close"
              class="close btn btn btn-outline-danger ms-3 btn-sm"
              data-testid="end-point-clear"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
        </div>
        <div
          class="d-flex flex-row mt-4"
        >
          <div
            class="me-3 d-flex flex-column align-items-center"
          >
            <button
              class="btn btn-outline-success"
              data-testid="sidebar-submit"
              disabled=""
            >
              <span />
            </button>
            <p
              class="text-success"
            >
              Search
            </p>
          </div>
          <div
            class="me-3 d-flex flex-column align-items-center"
          >
            <button
              class="btn btn-outline-danger"
              data-testid="sidebar-cancel"
              type="button"
            >
              <span />
            </button>
            <p
              class="text-danger"
            >
              Reset
            </p>
          </div>
        </div>
      </div>
      <div
        class="toggle-container"
        data-testid="sidebar-toggle"
      >
        <span />
      </div>
    </div>
  </form>
</div>
`;
